WPMedia/fusion-news-theme-blocks
# PEN-1439: Primary color option for nav bar color

## Description
Added logic for new blocks themes parameter, navBarBackground.  If navBarBackground is "primary-color" then that sites primary color is used as the nav bar background color.  The text color of the nav bar is black if the navColor is Light and white if the nav color is dark.  If no navBarBackground is specified for a site then that site's configuration for navColor is respected.

## Jira Ticket
- [PEN-1439](https://arcpublishing.atlassian.net/browse/PEN-1439)

## Acceptance Criteria
A new Theme Setting called navBarBackground is introduced 

If the value of navBarBackground is primary-color, then the customer’s primary color should be used as the background color for the navigation bar 

If navBarBackground is absent or set to any value other than primary-color, then the default color for dark or light mode should be used (black or white respectively) 

Tests and documentation are updated to cover this functionality 

## Test Steps
Add a nav bar Navigation – Arc Block @wpmedia/header-nav-block/navigation to a page.  Verify the follow scenarios:

1. If a site has  navBarBackground in blocks.json, then the primary color specified for that site is used to as the nav bar background color. 
A.  If the site navColor is light then text color on the nav bar is black.
B.  If the site navColor is dark then text color on the nav bar is white.

2. If a site has  navBarBackground in blocks.json, then that sites values for navColor is respected:
A. If navColor is light, then the nav bar will have white background with black text.
B If navColor is dark, then the nav bar will have black background with white text.


## Effect Of Changes
### Before
Only navColor is available as an option to configure the nav bar background color and text color (Light or Dark)
<img width="791" alt="Screen Shot 2020-11-20 at 5 13 00 PM" src="https://user-images.githubusercontent.com/2664083/99855294-872c3280-2b54-11eb-870a-457e5a2ac8e2.png">
<img width="1053" alt="Screen Shot 2020-11-20 at 5 19 09 PM" src="https://user-images.githubusercontent.com/2664083/99855296-872c3280-2b54-11eb-8178-c52b92f6f3c5.png">


### After
navBarBackground as well as navColor are available as options to allow for finer granularity in configurations for nav bar background and text color.

<img width="1037" alt="Screen Shot 2020-11-19 at 3 05 47 PM" src="https://user-images.githubusercontent.com/2664083/99854675-47188000-2b53-11eb-8d8d-051a8dd62426.png">

<img width="1095" alt="Screen Shot 2020-11-19 at 3 09 47 PM" src="https://user-images.githubusercontent.com/2664083/99854695-51d31500-2b53-11eb-8c6a-d32e41d5575e.png">
<img width="398" alt="Screen Shot 2020-11-19 at 3 09 55 PM" src="https://user-images.githubusercontent.com/2664083/99854700-55ff3280-2b53-11eb-9607-a8e2e4118012.png">
<img width="577" alt="Screen Shot 2020-11-19 at 3 10 03 PM" src="https://user-images.githubusercontent.com/2664083/99854708-5a2b5000-2b53-11eb-92d1-2bb9502fb84b.png">


## Dependencies or Side Effects
New blocks.json Pr is out for Fusion-news-Theme to allow testing this on corecomponents - otherwise there will be no sites that have the new navBarBackground parameter for determining nav bar background color.

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x ] Confirmed all the test steps above are working
- [ x] Confirmed there are no linter errors
- [ x] Confirmed this PR has reasonable code coverage
  - [ x] Confirmed this PR has unit test files
  - [x ] Ran `npm test`, made sure all tests are passing
  - [ x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x ] Confirmed relevant documentation has been updated/added.
